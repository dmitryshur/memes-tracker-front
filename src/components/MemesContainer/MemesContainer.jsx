import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { getChosenSection } from "../../selectors/visuals/visuals";
import "./MemesContainer.scss";

export const MemesContainer = props => {
  const matchingSectionOpened = props.chosenSection === props.section;
  const classList = `memes-container memes-container-${props.section} ${
    matchingSectionOpened ? "opened" : "closed"
  }`;
  return <div className={classList} />;
};

const mapStateToProps = state => ({
  chosenSection: getChosenSection(state)
});

MemesContainer.propTypes = {
  chosenSection: PropTypes.string
};

MemesContainer.defaultProps = {
  chosenSection: ""
};

export default connect(mapStateToProps)(MemesContainer);
