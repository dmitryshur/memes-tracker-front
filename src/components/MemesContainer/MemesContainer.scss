$memes-newest-bg: #34495e;
$memes-liked-bg: #ecf0f1;

.memes-container {
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 0;
  transition-property: transform;
  transition-duration: 1s;

  &.memes-container-newest {
    background-color: $memes-newest-bg;
    left: 0;
    transform: translateX(-100%);

    &.opened {
      transform: translateX(-20%);
    }
  }

  &.memes-container-liked {
    background-color: $memes-liked-bg;
    right: 0;
    transform: translateX(100%);

    &.opened {
      transform: translateX(20%);
    }
  }
}
