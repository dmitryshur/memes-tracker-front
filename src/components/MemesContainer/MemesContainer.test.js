import React from "react";
import { shallow } from "enzyme";
import { MemesContainer } from "./MemesContainer";

test("testing the rendering of the MemesContainer component of the liked section when opened", () => {
  const wrapper = shallow(<MemesContainer section="liked" chosenSection="liked" />);
  expect(wrapper).toMatchSnapshot();
});

test("testing the rendering of the MemesContainer component of the liked section when closed", () => {
  const wrapper = shallow(<MemesContainer section="liked" chosenSection="newest" />);
  expect(wrapper).toMatchSnapshot();
});

test("testing the rendering of the MemesContainer component of the newest section when opened", () => {
  const wrapper = shallow(<MemesContainer section="newest" chosenSection="newest" />);
  expect(wrapper).toMatchSnapshot();
});

test("testing the rendering of the MemesContainer component of the newest section when closed", () => {
  const wrapper = shallow(<MemesContainer section="newest" chosenSection="liked" />);
  expect(wrapper).toMatchSnapshot();
});
