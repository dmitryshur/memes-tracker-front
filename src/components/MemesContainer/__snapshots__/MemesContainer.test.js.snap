// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of the MemesContainer component of the liked section when closed 1`] = `
<div
  className="memes-container memes-container-liked closed"
/>
`;

exports[`testing the rendering of the MemesContainer component of the liked section when opened 1`] = `
<div
  className="memes-container memes-container-liked opened"
/>
`;

exports[`testing the rendering of the MemesContainer component of the newest section when closed 1`] = `
<div
  className="memes-container memes-container-newest closed"
/>
`;

exports[`testing the rendering of the MemesContainer component of the newest section when opened 1`] = `
<div
  className="memes-container memes-container-newest opened"
/>
`;
